kartothek/utils/repartition.py

```diff
@@ -0,0 +1,54 @@
+import math
+from typing import Optional
+
+import dask
+import dask.dataframe as dd
+
+import kartothek.io.dask.dataframe as ktk_ddf
+from kartothek.core.factory import DatasetFactory
+from kartothek.serialization import PredicatesType
+from kartothek.utils.optimize import optimize
+
+
+def calculate_repartition_ratio(
+    dataset_factory: DatasetFactory,
+    table_name: str,
+    predicates: Optional[PredicatesType] = None,
+    *,
+    target_nrows_per_partition: int,
+    dataset_metadata_frac: float,
+) -> dd.core.Scalar:
+    """
+    Calculate a `repartition_ratio` according to the dataset input and `target_nrows_per_partition` (the desired
+    number of rows of each dataframe partition).
+    """
```

Comment:
Can you please add a more thorough description here, i.e. parameter docs.

Reply:
Updated. Please verify.